cow_instrument/DetectorInfo.h

```diff
@@ -40,27 +40,45 @@ template <typename InstTree> class DetectorInfo {
 
   bool isMonitor(size_t detectorIndex) const;
 
-  void initL2();
+  double l2(size_t detectorIndex) const;
 
-  void initL1();
+  Eigen::Vector3d position(size_t componentIndex) const;
 
-  double l2(size_t detectorIndex) const;
+  Eigen::Vector3d positionDetector(size_t detectorIndex) const;
```

Comment:
👍 
The only reason (I remember) for moving *arbitrary* components via `DetectorInfo` was that we need to trigger updates of the L2 tables, but probably there is a way to do this with the changes you suggested above.